src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,544 @@
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Create Pcd8544
+        /// </summary>
+        /// <param name="dataCommandPin">The data command pin.</param>
+        /// <param name="spiDevice">The SPI device.</param>
+        /// <param name="resetPin">The reset pin. Use a negative number if you don't want to use it</param>
+        /// <param name="backlightPin">The pin back light</param>
+        /// <param name="gpioController">The GPIO Controller.</param>
+        /// <param name="shouldDispose">True to dispose the GPIO controller</param>
+        public Pcd8544(int dataCommandPin, SpiDevice spiDevice, int resetPin = -1, int backlightPin = -1, GpioController? gpioController = null, bool shouldDispose = true)
+        {
+            if (dataCommandPin < 0)
+            {
+                throw new ArgumentException($"{nameof(dataCommandPin)} must be a valid pin number");
+            }
+
+            _dataCommandPin = dataCommandPin;
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+            _shouldDispose = gpioController == null || shouldDispose;
+            _controller = gpioController ?? new();
+            _resetPin = resetPin;
+            if (resetPin >= 0)
+            {
+                _controller.OpenPin(resetPin, PinMode.Output);
+                _controller.Write(resetPin, PinValue.Low);
+                // Doc says at least 100 ns
+                Thread.Sleep(1);
+                _controller.Write(resetPin, PinValue.High);
+            }
+
+            _backlightPin = backlightPin;
+            if (backlightPin >= 0)
+            {
+                _controller.OpenPin(backlightPin, PinMode.Output);
+                _controller.Write(backlightPin, PinValue.Low);
+            }
+
+            _controller.OpenPin(_dataCommandPin, PinMode.Output);
+
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            _bias = 4;
+            _temperature = ScreenTemperature.Coefficient0;
+            _contrast = 0x30;
+            _enabled = true;
+            // Extended function, contrast to 0x30, temperature to coef 0, bias to 4, Screen to normal display power on, display to normal mode
+            SpiWrite(false, new byte[] { (byte)(FunctionSet.PowerOn | FunctionSet.ExtendedMode), (byte)(0x80 | _contrast), (byte)ScreenTemperature.Coefficient0, (byte)(0x10 | _bias), (byte)FunctionSet.PowerOn, (byte)DisplayControl.NormalMode });
+            Clear();
+            Draw();
+        }
+
+        #region properties
+
+        /// <summary>
+        /// Enable the screen
+        /// </summary>
+        public bool Enabled
+        {
+            get => _enabled;
+            set
+            {
+                _enabled = value;
+                byte enab = (byte)(_enabled ? FunctionSet.PowerOn : FunctionSet.PowerOff);
+                SpiWrite(false, new byte[] { enab });
```

Comment:
consider span

Reply:
... or maybe an overload that takes a single byte. Using SpiDevice.WriteByte() would not require any object instantiation.

Reply:
I was clearly a bit reluctant to create Span and then stackalloc and then put everything in there for 1 or 2 bytes. I understand the logic with the stack allocation but still.
Any advantage to use Span? 
And yes @pgrawehr , that would help to have a WriteByte(), there are quite some scenarios like this.

Reply:
And @krwq my primitive internal function takes byte[], not Span. I can always adjust but does that makes sense?

Reply:
there is less GC pressure overall. It might affect larger apps (it's about many tiny allocations all over not about this particular one). I do not think we currently have a need for super good perf. I'm ok with this as is and changing that in the future when we have a specific perf goals (that's why I wrote "consider" in the first place, nothing blocking here)

Reply:
OK, moved all to Span<byte>, thanks for the explanation, that helps.